src/frontend/src/app/pages/proposal-review-edit/proposal-review-edit.component.ts

```diff
@@ -334,4 +407,27 @@ export class ProposalReviewEditComponent implements OnInit {
     matchesDescription.updateValueAndValidity();
     summary.updateValueAndValidity();
   }
+
+  private onCommitShaChange(
+    commitSha: string | null | undefined,
+    commitForm: FormGroup<ReviewCommitForm>,
+  ): void {
+    if (!commitSha) {
+      return;
+    }
+
+    const result = extractCommitSha(commitSha);
+
+    if (!result) {
+      return;
+    }
+
+    commitForm.controls.id.setValue(result, { emitEvent: false });
+  }
+
+  private focusCommitShaInput(index: number): void {
+    setTimeout(() => {
```

Comment:
`setTimeout(() => {}, 0)` will wait until there are no tasks left on the JavaScript event loop before executing the callback, this makes sure that the element is ready to receive focus.